### DISCUSSION TITLE
Some users see information they shouldn't see in PNP-Search result webpart

### DISCUSSION THREAD

**Discussion by anditee** [microsoft-search/pnp-modern-search#3873] [Q&A]
**Version used**
4.12.2

**Describe the bug**
We are using the webpart to enable search for different articles (sitepages) within our sharepoint site. The solution works fine for almost all of our customers and all our users in the company.

Some of the customers experience a weird error depending issues with the results shown. At first glance we thought about an issue regarding the access rights of the user itself but testing it in SearchQueryTool by Microsoft doesn't show the same results as what we are experiencing within the webpart results on using the same user.

Query Text used:
{searchTerms}* 
ContentTypeId:{0x0101009D1CB255DA76424XXXXXXXXXXXXX...B204094B*} ContentClass:"STS_ListItem_WebPageLibrary"

What user with the error gets shown:
He can see everything for example other users names and e-mail adresses, draft versions, files he has only read access to. (about 70 files and all users)

What user without the error gets shown:
The corresponding documents with the right contentTypeID and he is able to filter (about 50 files)

**To Reproduce**
The only thing we noticed is that its on some users regardless if guest user or tenant user or admin.

**Expected behavior**
The experience should be equal for every user and only include the right content type. Furthermore the query text should apply filters to the search.

**Desktop (please complete the following information):**
 - all modern browsers

**Additional context**
Customer get access via cross tenant sync

**Comment by kasperbolarsen** [comment]
I assume you have tested if the user that are "seeing too much" actually can see the content,  following the link to the content? Getting an Access Denied?

**Comment by anditee** [comment]
you are right for example if he clicks on users which he sees (from another customer) which shouldn't be shown he gets an access denied.

**Comment by wobba** [comment]
Whatever is returned in search a particular user has access to, and you should be able to see the same data in Microsoft 365 search experiences as well. The issue is likely too much access given on content you see, or that access control has not updated after you changed it - which would allow items in search but clicking them would give access denied. I would hit "reindex" on the content in question and see if it's resolved in 24-72h. If not, log a support ticket with Microsoft thru official channels.

**Comment by kasperbolarsen** [comment]
And if you have to log a support ticket please remember to create a scenario using classic search or Microsoft search.  I have seen multiple cases where the support engineer rejected the ticket as soon as PnP Modern Search was mentioned.

**Comment by anditee** [comment]
Okay thanks for your answers. I totally agree with you guys. 

But why does the filter in the webpart doesn't apply? Shouldn't the information be filtered with the given queryText? The querytext works fine on the most of the clients and works also fine on users which can see everything within SearchQueryTool but not in the webpart.

Thanks for your replies :)

**Reply by kasperbolarsen** [reply]
@anditee has this been resolved,  and what was the root cause?

**Reply by anditee** [reply]
@kasperbolarsen so we found the issue our self. The problem is when you are using curly brackets within the filters they wont apply we fixed it by using quotation marks. But the users still get indexed this is a problem we couldn't fix by now. The users did show up in the first place because the filters didnt apply propperly.

If you use curly brackets within a post call the error happens in the get version it wont happen. And on some users there is no issue on the post query if you use them also.
